### PR TITLE
[IDE] Disable constraint system diagnostics during code completion

### PR DESCRIPTION
**Pull request by rintaro**
It seems disabling diagnostics doesn't introduce any regression.

rdar://problem/50679428

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/Sema/TypeCheckConstraints.cpp [hunk 1]
```diff
@@ -2175,7 +2176,8 @@ Type TypeChecker::typeCheckExpressionImpl(Expr *&expr, DeclContext *dc,
   // Construct a constraint system from this expression.
   ConstraintSystemOptions csOptions = ConstraintSystemFlags::AllowFixes;
 
-  if (options.contains(TypeCheckExprFlags::SuppressDiagnostics))
+  if (options.contains(TypeCheckExprFlags::SuppressDiagnostics) ||
+      DiagnosticSuppression::isEnabled(Diags))
```
Comment (slavapestov) [inline lib/Sema/TypeCheckConstraints.cpp]: @xedin Do we really need both mechanisms?

Comment (xedin) [reply lib/Sema/TypeCheckConstraints.cpp]: @rintaro Shouldn't enabled diagnostic suppression mean SuppressDiagnostics flag while forming constraint system?

Comment (rintaro) [reply lib/Sema/TypeCheckConstraints.cpp]: @xedin 
Currently, `DiagnosticSuppression` does nothing related to `TypeCheckExprFlags::SuppressDiagnostics`.

@slavapestov 
We can drop `TypeCheckExprFlags::SuppressDiagnostics` and consolidate it to `DiagnosticSuppression`.
Currently, we have only 2 usage of `TypeCheckExprFlags::SuppressDiagnostics`, and one of them is already under `DiagnosticSuppression`.

Comment (xedin) [reply lib/Sema/TypeCheckConstraints.cpp]: @rintaro I think `DiagnosticSuppression::isEnabled` returning true should correspond to `ConstraintSystemFlags::SuppressDiagnostics` flag set in `ConstraintSystem` so we only have one source of input about this in the solver.

Comment (rintaro) [reply lib/Sema/TypeCheckConstraints.cpp]: I added another commit to consolidate `TypeCheckExprFlags::SuppressDiagnostics` with `DiagnosticSuppression`. Is this what you are suggesting?

Comment (xedin) [reply lib/Sema/TypeCheckConstraints.cpp]: Yes, I think the way you changed it is okay, I just how that we wouldn't need to decouple it again later...

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/Sema/TypeCheckConstraints.cpp (hunk 1 above)
Comment (slavapestov) [reply lib/Sema/TypeCheckConstraints.cpp]: I'm all in favor of consolidating them. I don't have an opinion on which one is 'better', I just think it's confusing when there are two ways of doing the same thing :)

File: lib/Sema/TypeChecker.h [hunk 2]
```diff
@@ -233,9 +233,6 @@ enum class TypeCheckExprFlags {
   /// that we force for style or other reasons.
   DisableStructuralChecks = 0x02,
 
-  /// Set if the client wants diagnostics suppressed.
-  SuppressDiagnostics = 0x04,
```
Comment (slavapestov) [inline lib/Sema/TypeChecker.h]: Yay!

**Comment by nathawes** [comment]
@swift-ci Please Build Toolchain macOS Platform

**Comment by rintaro** [comment]
@swift-ci Please Sourcekit Stress test

**Comment by rintaro** [comment]
Added revert commit for #25922.

**Comment by rintaro** [comment]
@swift-ci Please Sourcekit Stress test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
Superseded by #26888

**Pull request closed without merging**